plugins/modules/vios.py

```diff
@@ -478,6 +493,68 @@ def installVios(module, params):
 
     return changed, vios_property, warn_msg
 
+def installViosUsingDisk(module, params):
+    hmc_host = params['hmc_host']
+    hmc_user = params['hmc_auth']['username']
+    password = params['hmc_auth']['password']
+    vios_iso = params['vios_iso']
+    image_dir = params['image_dir']
+    vios_IP = params['vios_IP']
+    gateway_ip = params['gateway_ip']
+    subnet_mask = params['subnet_mask']
+    mac_addr = params['mac_addr']
+    system_name= params['system_name']
+    vios_name = params['vios_name']
+    profile = params['profile']
+    label = params['label']
+    # logger.debug(vios_iso, image_name, vios_IP, gateway_ip, subnet_mask, mac_addr, system_name, vios_name, profile, label)
+    validate_parameters(params)
+    hmc_conn = HmcCliConnection(module, hmc_host, hmc_user, password)
+    hmc = Hmc(hmc_conn)
+    
+    try:
+        if mac_addr:
+            logger.debug("hi if")
+            hmc.installOSFromDisk(vios_iso, image_dir, vios_IP, gateway_ip, subnet_mask, mac_addr, system_name, vios_name, profile, label)
+        else:
+            logger.debug("hi else")
+            dvcdictlt = hmc.fetchIODetailsForNetboot(hmc_host, gateway_ip, vios_IP, vios_name, profile, system_name, subnet_mask)
```

Comment:
make sure that we handle the error cases when System is not connected to HMC or System is in powered off state